modules/data.land/R/extract_SM_CDS.R

```diff
@@ -0,0 +1,82 @@
+#' Extract CDS soil moisture data for the SDA workflow.
+#'
+#' @param site_info Bety list of site info including site_id, lon, and lat.
+#' @param time.points A vector contains each time point within the start and end date.
+#' @param in.path physical paths to where the unziped soil moisture files are downloaded.
+#' @param out.path Where the final CSV file will be stored.
+#' @param allow.download Flag determine if we want to automatic download files if they are not available.
+#' @param search_window search window for locate available soil moisture values.
+#'
+#' @return A data frame containing soil moisture and sd for each site and each time step.
+#' @export
+#' 
+#' @examples
+#' @author Dongchen Zhang
+#' @importFrom dplyr %>%
+extract_SM_CDS <- function(site_info,
+                           time.points, 
+                           in.path, 
+                           out.path = NULL, 
+                           allow.download = TRUE,
+                           search_window = 10){
+  #find downloaded files and the corresponding dates.
+  ncfiles <- base::list.files(in.path, full.names = T)
+  available.dates <- ncfiles %>% purrr::map(function(file){
+    base::strsplit(x = file, split = ".", fixed = T)[[1]][2]
+  }) %>% unlist %>% as.Date
+  #loop over time points.
+  results <- data.frame()
+  for (i in seq_along(time.points)) {
+    #generate dates within search window.
+    dates <- c(seq(lubridate::date(time.points[i]), by = "1 day", length.out = search_window))
+    dates.ind.download <- which(!dates%in%available.dates)
+    dates.ind.exist <- which(dates%in%available.dates)
+    #if there is no nc file downloaded or within the search window.
+    if (length(dates.ind.download) > 0) {
+      PEcAn.logger::logger.info(paste("The nc files for the following dates were not found.", 
+                                      paste(dates[dates.ind.download], collapse = ", ")))
+      #try download if allowed.
+      if (allow.download) {
+        PEcAn.logger::logger.info("Try download from cds server.")
+        ncfile <- c(ncfiles[dates.ind.exist], PEcAn.data.land::download.SM_CDS(in.path, dates[dates.ind.download])) %>% sort()
+      } else {
+        PEcAn.logger::logger.severe("The download is not enabled, skip to the next time point.")
```

Comment:
logger.severe won't skip to the next time point, it will kill the whole R job

Reply:
Replaced it with `logger.info`.